libcudacxx/test/public_headers/CMakeLists.txt

```diff
@@ -0,0 +1,39 @@
+# For every private public header build a TU that directly includes it
+# without anything else to verify whether its properly modular.
+# We also want to check whether all potential fallbacks are actually valid,
+# even if we never exercise them in the current support matrix
+add_custom_target(libcudacxx.test.public_headers)
+
+# Grep all public headers
+file(GLOB public_headers
+  LIST_DIRECTORIES false
+  RELATIVE "${libcudacxx_SOURCE_DIR}/include"
+  CONFIGURE_DEPENDS
+  "${libcudacxx_SOURCE_DIR}/include/cuda/*"
+  "${libcudacxx_SOURCE_DIR}/include/cuda/std/*"
+)
+
+# We must exclude those headers that not supported due to CUDA_ARCHITECTURES
+if (false)
```

Comment:
Can somebody that knows how to cmake tell me how I can determine whether we are at least SM70?

Reply:
Just to be clear, you only want to compile this for sm70+?

Reply:
Thanks to @robertmaynard I found an elegant way to solve this 😻